src/DotLiquid/StandardFilters.cs

```diff
@@ -827,10 +913,27 @@ public static IEnumerable Uniq(object input)
         /// </summary>
         /// <param name="context">The DotLiquid context</param>
         /// <param name="input">Input to be transformed by this filter</param>
-        public static double Abs(Context context, object input)
+        [LiquidFilter(MinVersion = SyntaxCompatibility.DotLiquid24)]
+        public static object Abs(Context context, object input)
         {
-            Double n;
-            return Double.TryParse(input.ToString(), NumberStyles.Number, context.CurrentCulture, out n) ? Math.Abs(n) : 0;
+            dynamic value = NumericConverter.CoerceToNumericType(input, context.FormatProvider, 0);
+            if (value >= 0)
+            {
+                return value;
+            }
+
+            // Try multiplying my -1 to get the absolute value.
+            dynamic minusOne = Convert.ChangeType(-1, value.GetType());

```

Comment:
I like the new method as it avoids type casting back and forth, but why not
```suggestion
            int minusOne = -1;
```
per MS you can combine math ops with int + float just not two differing types of float...